### ISSUE TITLE
useLadleState doesn't change values upon state changes

### ISSUE THREAD

**Issue by viddo** [tajo/ladle#165]
**Describe the bug**

I'd like to be able to style stories per light vs dark mode. I found `useLadleState` hook in the public exports, but it seems stuck with the initial values as defined in the context file: https://github.com/tajo/ladle/blob/c229fd3ef187f12a260e5203c4dcdb88d7152a7e/packages/ladle/lib/app/src/context.ts#L13-L20

Can observe that the hook triggers (the console.log), it's just that the _values_ aren't updated.

**Reproduction**

- https://stackblitz.com/edit/ladle-epoh4u?file=vite.config.mjs,src%2Fbasic.stories.tsx
  - _or:_
  ```tsx
  import { useLadleState } from "@ladle/react";
  
  export const World = () => {
    const [globalState] = useLadleState();
    console.log('globalState', globalState);
    return (
      <p style={{ color: globalState.theme === "light" ? "#ace" : "#fabada" }}>
        Hey! <pre>{JSON.stringify(globalState, null, 2)}</pre>
      </p>
    );
  };
  ```

- Use UI to change Ladle state, e.g. toggle dark mode, RTL, etc.

---

BTW Stackblitz boilerplate didn't work out of the box, needed to add a `vite.config.mjs` and override default server.open value for `npm run dev` to not fail:
```mjs
export default {
  server: {
    open: 'none'
  }
}
```

**Comment by tajo** [comment]
> BTW Stackblitz boilerplate didn't work out of the box

Fixed that, thanks.

Frankly, not sure why

```js
const [globalState] = useLadleState();
console.log('globalState', globalState);
```

doesn't update. Interestingly, it works inside of the project's monorepo (if you try it inside of `example` project).

Also, it's not a documented API - it probably should be. It might be also better to change it a bit and expose it directly as `useLadleContext`.

**Comment by viddo** [comment]
> Interestingly, it works inside of the project's monorepo (if you try it inside of example project).

You're referring to this, right? https://github.com/tajo/ladle/blob/master/CONTRIBUTING.md#developing 

> Also, it's not a documented API - it probably should be. 

Ah yeah, indeed. I should probably have started my issue by asking: what would be the official way to get the Ladle dark mode state in stories? 😅

> It might be also better to change it a bit and expose it directly as useLadleContext.

If you already have plans to expose this kind of state through some other API I can wait for that and try it out when it's ready? Otherwise I could try to troubleshoot it further to see if I can find the culprit (although later on, I'm on traveling foot til next week).

**Comment by tajo** [comment]
> expose this kind of state through some other API

Nah, I like using the hook + context to get access to the internal state and `dispatch` function. It's very flexible since it gives you the full access to everything. 

I just need to figure out why it doesn't work.

**Comment by tajo** [comment]
Also, it doesn't work only in dev (server) mode.

**Comment by tajo** [comment]
Found a fix, released as `v1.1.2`.

Also added some [docs](https://ladle.dev/docs/providers#ladle-context) for `useLadleContext` but keeping it marked as experimental for now.
